addons/account/models/account_move_line.py

```diff
@@ -746,7 +746,7 @@ def _compute_amount_residual(self):
             line.reconciled = (
                 comp_curr.is_zero(line.amount_residual)
                 and foreign_curr.is_zero(line.amount_residual_currency)
-                and (line.matched_debit_ids or line.matched_credit_ids)
+                and line.move_id.state not in ('draft', 'cancel')
```

Comment:
Please, I would like you avoid forcing this. All this began in https://github.com/odoo/odoo/pull/92666. As @william-andre said, the reconciled flag should be independent of the state, it only should check if there are account.partial.reconcile records for it.

Reply:
@MiquelRForgeFlow Hmmm ok but I can't let you force the orm to fetch `matched_debit_ids`/`matched_credit_ids` for performance matters

Reply:
Edit: If the move is draft/cancel, there is no reconciliation since the move needs to be posted to be reconciled. So what's the problem with the current code?

Reply:
Sometimes, someone may want put in draft an already posted reconciled move and don't want the reconciliation be unreconciled.

Reply:
> If the move is draft/cancel, there is no reconciliation since the move needs to be posted to be reconciled

Also, some client may want to be able to reconcile draft moves.

Reply:
@MiquelRForgeFlow What about my last commit? Do we even need to check for matched_debit_ids/matched_credit_ids?

Reply:
Ok, seems better to me, thanks. This part of code was refactored in https://github.com/odoo/odoo/commit/5883f0a84cd3cd2330cc5ff34261252dacd8c5b9 and probably it doesn't need my old fix now. When I forward ported the fix, I was just too focused in the forward port and didn't realize that. I will test a bit, but I am not the accounting expert here hehe